cypress/e2e/signUp.cy.js

```diff
@@ -1,12 +1,52 @@
 /// <reference types='cypress' />
 /// <reference types='../support' />
 
+import SignUpPageObject from "../support/pages/signUp.pageObject";
+import HomePageObject from "../support/pages/home.pageObject";
+
+const signUpPage = new SignUpPageObject();
+const homePage = new HomePageObject();
+
 describe('Sign Up page', () => {
+  let user;
+  let wrongUser;
+
   before(() => {
+    cy.task('db:clear');
+    cy.task('generateUser').then((generateUser) => {
+      user = generateUser;
+    });
+    cy.task('generateWrongUser').then((generateWrongUser) => {
+      wrongUser = generateWrongUser;
+    });
+  });
+
+  it('should provide an ability to sign up with valid credentials', () => {
+    signUpPage.visit();
+    signUpPage.typeUsername(user.username);
+    signUpPage.typeEmail(user.email);
+    signUpPage.typePassword(user.password);
+
+    signUpPage.clickSignUpBtn();
+
+    homePage.assertHeaderContainUsername(user.username);
+  });
+
+  it('should not provide an ability to register with wrong email', () => {
+    cy.register(wrongUser.email, user.username, user.password);
```

Comment:
Instead of directly calling `cy.register`, consider using `signUpPage` methods to simulate user input for the wrong email case. This will better mimic the user interaction with the sign-up form.